File: clojure/prime-factors/src/prime_factors.clj
(ns prime-factors)

(defn of [n]
  (loop [of-n n
         div 2
         factors []]
    (cond
      (= of-n 1)             factors
      (zero? (mod of-n div)) (recur (quot of-n div) div (conj factors div))
      :else                  (recur of-n (inc div) factors))))

File: ruby/prime-factors/prime_factors.rb
module PrimeFactors
  def self.of(n)
    factors = []
    d = 2
    while n > 1
      q, m = n.divmod d
      if m.zero?
        factors << d
        n = q
      else
        d += 1
      end
    end
    factors
  end
end
